kotest-assertions/kotest-assertions-json/src/jsTest/kotlin/io.kotest.assertions.json/JsonLiteralsTest.kt

```diff
@@ -63,7 +75,69 @@ class JsonLiteralsTest : FunSpec(
                actual:
                true
             """.trimIndent()
-         )
+            )
+         }
+      }
+
+      context("CompareMode.Exact requires same format for numbers") {
+         infix fun String.shouldExactlyEqualJson(expected: String) = this.shouldEqualJson(expected, CompareMode.Exact)
+
+         test("comparing float and exponent") {
+            shouldFail {
+               "10.0" shouldExactlyEqualJson "1e1"
+            }.shouldHaveMessage(
+               """
+               The top level expected 1e1 but was 10.0
+
+               expected:
+               10.0
+
+               actual:
+               10.0
```

Comment:
This looks/feels a bit odd.

What happens is the json document is parsed to `JsonElement` and then encoded back to string again. (`JsonMatchers.kt:73-77`). I think it would be worthwhile  to roll our own prettification of JSON-document that prevents parsing of the values.. Shall I file an issue for it?